src/components/MovieDetails/MovieDetails.spec.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+
+import { MovieDetails } from "./MovieDetails";
+import { MovieContext } from "@context/MovieContext";
+import { moviesList } from "@shared/constants";
+
+describe("MovieDetails", () => {
+  it("should be rendered correctly when a movie is selected", () => {
+    render(
+      <MovieContext.Provider value={{ selectedMovie: moviesList[0] } as any}>
+        <MovieDetails />
+      </MovieContext.Provider>,
+    );
+
+    const movieImage = screen.getByAltText(moviesList[0].name);
+    expect(movieImage).toBeInTheDocument();
+    expect(movieImage).toHaveAttribute("src", moviesList[0].imageUrl);
+
+    const movieName = screen.getByRole("heading", { level: 3 });
+    expect(movieName).toBeInTheDocument();
+    expect(movieName).toHaveTextContent(moviesList[0].name.toUpperCase());
+
+    const movieRating = screen.getByText(moviesList[0].rating.toString());
+    expect(movieRating).toBeInTheDocument();
+
+    const movieGenres = screen.getByText(moviesList[0].genres.join(", "));
+    expect(movieGenres).toBeInTheDocument();
+
+    const movieReleaseYear = screen.getByText(
+      moviesList[0].releaseYear.toString(),
+    );
+    const movieDuration = screen.getByText(moviesList[0].duration);
+    expect(movieReleaseYear).toBeInTheDocument();
+    expect(movieDuration).toBeInTheDocument();
+
```

Comment:
JFYI I think in production React project we do not check all elements. 1-2 would be enough. Mandatory to check element that show or hide for some certain conditions.

Reply:
Left only some test cases